daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -241,14 +252,14 @@ object Speedy {
     /** Compute a stack trace from the locations in the continuation stack.
         The last seen location will come last. */
     def stackTrace(): ImmArray[Location] = {
-      val s = new util.ArrayList[Location]
+      val s = ImmArray.newBuilder[Location]
```

Comment:
This change does not seem to be related to making things private. I assume `ImmArray` is expected to be more performant than `ArrayList` in this case?

Reply:
`ImmArray` builders are probably not as efficient as `ArrayList`. However in the current case the `ArrayList` is built and then directly converted in `ImmArray`